engine/access/rpc/rate_limit_test.go

```diff
@@ -117,8 +119,40 @@ func (suite *RateLimitTestSuite) SetupTest() {
 	block := unittest.BlockHeaderFixture()
 	suite.snapshot.On("Head").Return(block, nil)
 
-	rpcEngBuilder, err := NewBuilder(suite.log, suite.state, config, suite.collClient, nil, suite.blocks, suite.headers, suite.collections, suite.transactions, nil,
-		nil, suite.chainID, suite.metrics, 0, 0, false, false, apiRateLimt, apiBurstLimt, suite.me)
+	backend := backend.New(
```

Comment:
@peterargue I think in such tests it's enough to mock backend, instead of mocking every dependency of backend. I am not asking to change it, since it's a pattern which is used in many tests but could be a thing for future task to refactor access/observer tests for easier maintainability.